driver/driver.cpp

```diff
@@ -561,27 +568,52 @@ Routine Description:
     //
     // Connect a mouse class device driver to the port driver.
     //
-    case IOCTL_INTERNAL_MOUSE_CONNECT:
+    case IOCTL_INTERNAL_MOUSE_CONNECT: {
         //
         // Only allow one connection.
         //
         if (devExt->UpperConnectData.ClassService != NULL) {
             status = STATUS_SHARING_VIOLATION;
             break;
         }
-        
+
         //
         // Copy the connection parameters to the device extension.
         //
-         status = WdfRequestRetrieveInputBuffer(Request,
-                            sizeof(CONNECT_DATA),
-                            reinterpret_cast<PVOID*>(&connectData),
-                            &length);
-        if(!NT_SUCCESS(status)){
+        status = WdfRequestRetrieveInputBuffer(Request,
+            sizeof(CONNECT_DATA),
+            reinterpret_cast<PVOID*>(&connectData),
+            &length);
+        if (!NT_SUCCESS(status)) {
             DebugPrint(("WdfRequestRetrieveInputBuffer failed %x\n", status));
             break;
         }
-        
+
+        // taken from REGSTR_VAL_MAX_HCID_LEN defined in um/RegStr.h
+        const size_t POOL_SIZE = sizeof(wchar_t) * 1024;
+
+        auto pool = ExAllocatePoolWithTag(NonPagedPool, POOL_SIZE, 'AR');
+
+        if (!pool) {
+            DebugPrint(("RA - failed to allocate pool for hwid list\n"));
+        }
+        else {
+            RtlZeroMemory(pool, POOL_SIZE);
+
+            ULONG resultLen;
+            NTSTATUS tmp = WdfDeviceQueryProperty(hDevice, DevicePropertyHardwareID,
```

Comment:
there's a bsod here waiting to happen

`WdfDeviceQueryProperty` must be called at IRQL = PASSIVE_LEVEL, which is not guaranteed in this function

will fix